### PR TITLE
feat: Add integrations class

### PR DESCRIPTION
**Pull request by agrohs**
## Status

* ✅ Ready

## Type

* ### Feature
  Added Integrations Endpoint to js-sdk
* ### Fix
  Adds new Endpoint, but also fixed fixes Collections endpoint in the moltin.d.ts file a bug
* ### Docs
  Documentation only changes
* ### Style
  Changes that **do not** affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
* ### Refactor
  A code change that neither fixes a bug nor adds a feature
* ### Performance
  A code change that improves performance
* ### Test
  Did not commit tests as the dependency on where the tests would hit (requestbin/otherwise) and whether tests should check to see if integrations were receiving requests was TBD.
* ### Chore
  Changes to the build process or auxiliary tools and libraries such as documentation generation  

## Description

*Already making use of the Integrations endpoint to add/update/remove webhooks, but was not exposed in SDK. Added it in as a basic CRUDExtend endpoint type.*

## Dependencies

*Other PRs or builds that this PR depends on.*

## Issues

*A list of issues closed by this PR.*

* Fixes #

## Notes

*Also added an es-lint disable no-console into the Transacations Endpoint above the debug.warn*

### CODE REVIEW

**Comment by agrohs** [comment]
cc: @gje4

**Comment by ynnoj** [comment]
Hey @agrohs 👋 

Are you able to add some tests? Thanks!

**Comment by agrohs** [comment]
@ynnoj, just added basic tests

File: test/unit/integrations.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    nock(apiUrl, {
+      reqheaders: {
+        Authorization: 'Bearer: a550d8cbd4a4627013452359ab69694cd446615a'
+      }
+    })
+      .get('/integrations')
+      .reply(200, integrations)
+
+    return Moltin.Integrations.All().then(response => {
+      console.log(response)
```
Comment (ynnoj) [inline test/unit/integrations.js]: @agrohs Remove `console.log`.

File: test/unit/integrations.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      assert.lengthOf(response, 2)
+    })
+  })
+
+  // it('should return a single integration', () => {
```
Comment (ynnoj) [inline test/unit/integrations.js]: @agrohs Why is this commented out?

File: test/unit/integrations.js (hunk 1 above)
Comment (ynnoj) [reply test/unit/integrations.js]: @agrohs I'm still seeing the `console.log` and the test which is commented out.

**Comment by ynnoj** [comment]
@agrohs What are all the `package-lock.json` changes?

**Comment by agrohs** [comment]
> @agrohs What are all the `package-lock.json` changes?

They looked to me to just be regenerated SHA-1 hash's that got generated when npm installing/building locally, but nothing on my end desired for any changes in there. Let me take a quick peak at removing that from the PR/reverting as-is to previous version and ensure all still works fine. Will reply back and revert that commit if all good!

**Comment by ynnoj** [comment]
@agrohs Yeah, would prefer if there weren't any lockfile changes in the PR as we've not added/removed any dependencies.

**Comment by ynnoj** [comment]
@agrohs 👋 What's the status with this PR?

**Comment by agrohs** [comment]
Awaiting updates around tests if I recall? More than happy to contribute the code, but the process to try and help you guys became quite burdensome. Not sure if anyone on the team wants to help get this set?

www.uniquelyparticular.com

**Comment by agrohs** [comment]
> Looks like we’re missing a test for the Get method on the Class.
@ynnoj - updated unit tests, adding in Get method test

cc: @gje4

**Pull request merged**